ledger-core/v2/virtual/execute/execute.go

```diff
@@ -100,6 +102,7 @@ func (s *SMExecute) prepareExecution(ctx smachine.InitializationContext) {
 }
 
 func (s *SMExecute) Init(ctx smachine.InitializationContext) smachine.StateUpdate {
+	ctx.SetDefaultMigration(s.migrateDuringExecution)
```

Comment:
Not sure if it's right place.
If we don't start an execution - then we don't need to set "migrationHappened" and we need to stop executing (not sure about executing part, more like "increment counter part")

Reply:
Yes, move it to stepUpdatePendingCounters step